Server/src/main/java/net/simon987/server/game/World.java

```diff
@@ -103,6 +103,7 @@ public void update() {
             //Clean up dead objects
             if (object.isDead()) {
                 gameObjects.remove(object);
+                object.onDeadCallback();
```

Comment:
It does not really change anything but the callback in World.java could be called before the object is removed, as per your comment `Called before this GameObject is removed from the world`